File: SeleniumExercise/src/main/java/applicationpages/manager/CreateModalPage.java
package applicationpages.manager;

import applicationpages.BasePage;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class CreateModalPage extends BasePage {
	@FindBy(how = How.CSS, using = "[data-testid='submit-Create']")
	private WebElement createButton;

	@FindBy(how = How.CSS, using = "[data-testid='submit-Cancel']")
	private WebElement cancelButton;

	@FindBy(how = How.CSS, using = "[data-testid='inputDialog-TextField'] div input")
	private WebElement nameTextBox;

	public CreateModalPage(WebDriver driver) {
		super(driver);
		waitForElementClickable(createButton);
	}

	public void setName(String folderName) {
		nameTextBox.click();
		nameTextBox.sendKeys(folderName);
	}

	public boolean isCreateButtonEnabled() {
		return !Boolean.parseBoolean(createButton.getAttribute("aria-disabled"));
	}

	public ManagerPage clickCreateButton() {
		createButton.click();
		try {
			Thread.sleep(5000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
		return new ManagerPage(driver);
	}

	public ManagerPage clickCancelButton() {
		cancelButton.click();
		return new ManagerPage(driver);
	}
}

File: SeleniumExerciseC#/Pages/manager/CreateModalPage.cs
using OpenQA.Selenium;
using SeleniumExtras.PageObjects;

namespace Pages
{
    class CreateModalPage : BasePage
    {
        [FindsBy(How = How.CssSelector, Using = "[data-testid='submit-Create']")]
        private IWebElement createButton;

        [FindsBy(How = How.CssSelector, Using = "[data-testid='submit-Cancel']")]
        private IWebElement cancelButton;

        [FindsBy(How = How.CssSelector, Using = "[data-testid='inputDialog-TextField'] div input")]
        private IWebElement nameTextBox;

        public CreateModalPage(IWebDriver driver) : base(driver)
        {
            waitForElementClickable(createButton);
        }
        public void setName(String folderName)
        {
            nameTextBox.Click();
            nameTextBox.SendKeys(folderName);
        }

        public bool isCreateButtonEnabled()
        {
            return !bool.Parse(createButton.GetAttribute("aria-disabled"));
        }

        public ManagerPage clickCreateButton()
        {
            createButton.Click();
            try
            {
                Thread.Sleep(5000);
            }
            catch (ArgumentOutOfRangeException e)
            {
                Console.WriteLine(e.ToString());
            }
            return new ManagerPage(driver);
        }

        public ManagerPage clickCancelButton()
        {
            cancelButton.Click();
            return new ManagerPage(driver);
        }
    }
}